### ISSUE TITLE
dates in frontmatter are not sorted as dates

### ISSUE THREAD

**Issue by deltaskelta** [ChristopherBiscardi/gatsby-mdx#263]
**Describe the bug**

Using the [gatsby-blog-starter](https://github.com/gatsbyjs/gatsby-starter-blog) everything in the frontmatter that looks like a date is parsed as a date and it also has some functions that can be called as args on the graphql query field

![screen shot 2019-01-20 at 8 57 54 pm](https://user-images.githubusercontent.com/9424192/51438958-a3f84480-1cf6-11e9-9f94-14ccfbb6fefc.png)

**To Reproduce**

With my mdx setup, the dates come in as strings and are sorted as such so that 

`Fri, Jan, 01 2000 00:00:00 GMT` comes before `Sat, Jan 01 2019 00:00:00 GMT`

**Expected behavior**

I would expect the frontmatter to be parsed like `gatsby-remark` and treat the dates as dates

**Screenshots**

![screen shot 2019-01-20 at 9 09 25 pm](https://user-images.githubusercontent.com/9424192/51439015-b921a300-1cf7-11e9-8e48-864d4f3e1392.png)

**Additional context**

**Comment by ChristopherBiscardi** [comment]
I ran into this recently too. Need to look into how the inference is happening. I only see this snippet in the remark transformer -- https://github.com/gatsbyjs/gatsby/blob/fc2ff3503ac1ed32a4b7677b12edf4c4ea4fd050/packages/gatsby-transformer-remark/src/on-node-create.js#L21-L34

**Comment by deltaskelta** [comment]
Yeah it must be the graymatter parsing or something before that that parses the dates. I'll see if I can track it down

**Comment by deltaskelta** [comment]
So I think I got confused somewhere along the way. I could have sworn that I tried the date format `Mon, 03 Nov 2014 11:34:37 GMT` and had it parse while using gatsby-remark in my testing. I cannot make it happen now though so IDK what happened. 

I did find this though that says it is not a valid format for gatsby. I tried some of the following valid formats and they parse as dates with gatsby-mdx just fine

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/schema/types/type-date.js#L11
